### PR TITLE
Added new Audio extraction + small fixes

### PR DESCRIPTION
**Pull request by mdda**
I'm interested in the whole audio-book as dataset idea, so I've added a 'librivox' extractor (though it could be called 'audiofiles' since it'll handle all audio in a directory).

There are other fixes (mostly typos), and sorting out the matplotlib backend - which seems to be a fix for the async Tkinter Image crash problem (hasn't happened to me since I implemented the change).

Let me know if there's anything you'd like done differently...

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -633,11 +641,13 @@ def __train_step(phase, epoch, global_step, global_test_step,
...
     mask = mask[:, 1:, :]
 
-    # Apply model
+    # Apply model: Run the model in regular eval mode
     # NOTE: softmax is handled in F.cross_entrypy_loss
     # y_hat: (B x C x T)
+    
     y_hat = model(x, c=c, g=g, softmax=False)
-
+    #print("len(y_hat)", len(y_hat))
```
Comment (r9y9) [inline train.py]: remove

File: train.py [hunk 2]
```diff
@@ -692,14 +702,17 @@ def train_loop(model, data_loaders, optimizer, writer, checkpoint_dir=None):
                 ema.register(name, param.data)
     else:
         ema = None
+        
+    _tqdm = tqdm
+    #_tqdm = lambda x: x
```
Comment (r9y9) [inline train.py]: remove

File: train.py [hunk 3]
```diff
@@ -692,14 +702,17 @@ def train_loop(model, data_loaders, optimizer, writer, checkpoint_dir=None):
...
 
     global global_step, global_epoch, global_test_step
     while global_epoch < hparams.nepochs:
         for phase, data_loader in data_loaders.items():
             train = (phase == "train")
             running_loss = 0.
             test_evaluated = False
-            for step, (x, y, c, g, input_lengths) in tqdm(enumerate(data_loader)):
+            for step, (x, y, c, g, input_lengths) in _tqdm(enumerate(data_loader)):
```
Comment (r9y9) [inline train.py]: I am assuming this was just for debugging?

File: train.py [hunk 4]
```diff
@@ -718,6 +731,7 @@ def train_loop(model, data_loaders, optimizer, writer, checkpoint_dir=None):
                     print("[{}] Eval at train step {}".format(phase, global_step))
 
                 # Do step
+                #print("\nRunning epoch %5d, step %6d - %s : %s" % (global_epoch, step, phase, ('EVAL' if do_eval else 'TRAIN'), ))
```
Comment (r9y9) [inline train.py]: remove

File: train.py [hunk 5]
```diff
@@ -182,16 +185,18 @@ def __init__(self, data_root, **kwargs):
...
     1. Sort by lengths
     2. Pick a small patch and randomize it
-    3. Permutate mini-batchs
+    3. Permutate mini-batches
     """
 
     def __init__(self, lengths, batch_size=16, batch_group_size=None,
                  permutate=True):
         self.lengths, self.sorted_indices = torch.sort(torch.LongTensor(lengths))
+        # print(self.lengths, self.sorted_indices, batch_size, batch_group_size)
```
Comment (r9y9) [inline train.py]: remove

**Comment by mdda** [comment]
Your changes made as requested

**Comment by r9y9** [comment]
Thanks!

**Comment by imdatsolak** [comment]
@mdda, we are currently finalizing speech training-data based mostly on LibriVox + Project Gutenberg. It will be available around March 20 on our website. Languages are: German, Spanish, Italian, Russian, Ukrainian, English (Queen's), English (US). French is also WiP. After March 20, you will be able to download it from our website. The total duration of all of them is (v1.0): 741hrs (~82GB).
They are all sentence-aligned, cleaned, etc, etc. Maybe that will be helpful.

**Comment by mdda** [comment]
That sounds super-interesting.  I'm looking forward to playing with the data - excellent stuff!

**Pull request merged**